common/app/conf/Filters.scala

```diff
@@ -67,6 +67,20 @@ object SurrogateKeyFilter extends Filter with ExecutionContexts {
   }
 }
 
+object AmpFilter extends Filter with ExecutionContexts with implicits.Requests {
+  override def apply(nextFilter: (RequestHeader) => Future[Result])(request: RequestHeader): Future[Result] = {
+    if (request.isAmp) {
+      val domain = request.headers.get("Origin").fold("https://" + request.domain)(_)
```

Comment:
is this folding over an option? I had to google to work out what that even means :confounded: but apparently you could just just change to something like ```
val domain = request.headers.get("Origin").getOrElse("https://" + request.domain)
```